File: app/Modules/Holidays/Models/Holiday.php
<?php

declare(strict_types=1);

namespace App\Modules\Holidays\Models;

use Illuminate\Database\Eloquent\Model;

class Holiday extends Model
{
    protected $table = 'holidays';

    protected $fillable = [
        'country_code',
        'holiday_date',
        'name',
        'type',
        'is_active',
        'source',
        'created_by',
    ];

    protected $casts = [
        'holiday_date' => 'date',
        'is_active' => 'bool',
        'created_by' => 'int',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];
}

File: mobile/lib/models/holiday.dart
class Holiday {
  final int id;
  final String countryCode;
  final String holidayDate;
  final String name;
  final String type;
  final bool isActive;
  final String source;

  Holiday({
    required this.id,
    required this.countryCode,
    required this.holidayDate,
    required this.name,
    required this.type,
    required this.isActive,
    required this.source,
  });

  factory Holiday.fromJson(Map<String, dynamic> json) {
    return Holiday(
      id: (json['id'] as num?)?.toInt() ?? 0,
      countryCode: json['country_code'] as String? ?? 'ET',
      holidayDate: json['holiday_date'] as String? ?? '',
      name: json['name'] as String? ?? '',
      type: json['type'] as String? ?? 'public',
      isActive: json['is_active'] as bool? ?? true,
      source: json['source'] as String? ?? 'manual',
    );
  }
}
